src/treesheets_impl.h

```diff
@@ -1,36 +1,19 @@
 struct TreeSheetsScriptImpl : public ScriptInterface {
     Document *document = nullptr;
     Cell *current = nullptr;
-    Cell *lowestcommonancestor = nullptr;
+    bool docmodified = false;
 
     enum { max_new_grid_cells = 256 * 256 };  // Don't allow crazy sizes.
 
     void SwitchToCurrentDocument() {
         document = sys->frame->GetCurrentTab()->doc;
         current = document->root;
-        lowestcommonancestor = nullptr;
-    }
-
-    void AddUndoIfNecessary() {
-        if (!lowestcommonancestor) {
-            UpdateLowestCommonAncestor();
-        } else {
-            for (auto p = current; p; p = p->parent) {
```

Comment:
Actually, this was still useful code.. if a script generates thousands of changes, this code will make that into very few undo points.

Reply:
Ah yes, you are completely right.